src/crimson/osd/pg.cc

```diff
@@ -1004,13 +1005,27 @@ PG::with_locked_obc(const hobject_t &hobj,
     throw crimson::common::system_shutdown_exception();
   }
   const hobject_t oid = get_oid(hobj);
+  auto wrapper = [f=std::move(f), this](auto obc) {
+    if (obc->watchers.empty()) {
+      for (auto &[src, winfo] : obc->obs.oi.watchers) {
+        auto watch = crimson::osd::Watch::create(
+          obc, winfo, src.second, this);
+        watch->disconnect();
+        auto [it, emplaced] = obc->watchers.emplace(src, std::move(watch));
+        assert(emplaced);
+        logger().debug("added watch for obj {}, client {}",
+          obc->get_oid(), src.second);
+      }
+    }
+    return f(obc);
+  };
```

Comment:
Looks good! What do you think about moving it into a separate method?

Reply:
Done